### PR TITLE
[https://nvbugs/6252549][fix] beam search log probs slot index

### PR DESCRIPTION
**Pull request by mattwittwer**
## Summary by CodeRabbit

# Release Notes

* **Documentation**
  * Updated product branding from "TensorRT-LLM" to "TensorRT LLM"
  * Bumped version badge to stable release (1.0.0)
  * Updated benchmarking documentation with UCX-based KVCache transfer

* **New Features**
  * Added custom MoE routing kernel for improved expert selection performance
  * Enhanced disaggregated serving support with improved KVCache sizing calculations
  * Added LoRA tensor removal capability for better memory management

* **Improvements**
  * Optimized logits post-processing parameter handling
  * Refined batch slot indexing for log probability tracking
  * Improved LoRA configuration validation logic
  * Updated pre-compiled CUDA kernels

[Review Change Stack](https://app.coderabbit.ai/change-stack/NVIDIA/TensorRT-LLM/pull/14699?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

## Description

With beam_width > 1 and return_log_probs=true, TensorRT-LLM returns inconsistent output_log_probs for identical, repeated requests. The generated tokens and cum_log_probs are correct and stable across runs; only the per-step output_log_probs array is wrong, and it differs run-to-run depending on which decoder slot the request is assigned. Root cause: the beam-search gather kernel (cpp/tensorrt_llm/kernels/decodingKernels.cu, insertUnfinishedPathKernel) indexes the shared logProbsTiled buffer using the kernel block index bid (always 0 for a single-slot finalize) instead of the request's actual batch slot. It therefore reads slot 0's region regardless of which slot actually produced the data, picking up stale/again from other slots. 

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [ ] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Comment by fredricz-20070104** [comment]
@mattwittwer  You need to amend -s "commit message" to add DCO sign before merge.

**Review by jieli-matrix — CHANGES_REQUESTED** [review]
This is a community changes, we'd better figure out users' intuition for the PR(it's a workaround or new feat.) then consider merge.

**Review by ruodil — APPROVED** [review]

File: cpp/tensorrt_llm/kernels/decodingKernels.cu [hunk 1]
```diff
@@ -804,6 +813,17 @@ void gatherTree(DecodingOutput const& decodingOutput, DecodingInput const& decod
     bh.outputIdsUnfinish = bufferCast<TokenIdType>(*decodingOutput.ids);
     bh.parentIdsUnfinish = bufferCast<TokenIdType>(*decodingOutput.parentIds);
 
+    // logProbsTiled is the only joint (un-sliced) tensor read by the gather kernel.
+    // The kernel is launched with one block per batch entry (here, a single block since
+    // we are gathering one slot at a time), so blockIdx.x is always 0. To index the
+    // correct slot of the joint tensor, expose the actual slot via bh.batchSlots.
+    auto batchSlotsPtr = manager.gpu(ITensor::makeShape({1}), TRTDataType<SizeType32>::value);
+    {
```
Comment (sunnyqgg) [inline cpp/tensorrt_llm/kernels/decodingKernels.cu:821]: This creates a one-element batchSlots buffer unconditionally, so it is only safe when gatherTree is called with bh.nBatchSize == 1. However, gatherTree also has full-batch callers, and insertUnfinishedPathKernel indexes bh.batchSlots by bid for each launched block. For batchSize > 1, bh.batchSlots must either be nullptr for identity mapping, or contain batchSize entries with the real slot for each batch item. Otherwise bid > 0 will read past this one-element buffer.

**Review by sunnyqgg — APPROVED** [review]

**Review by nvpohanh — APPROVED** [review]

**Comment by mattwittwer** [comment]
This has been superseded by https://github.com/NVIDIA/TensorRT-LLM/pull/15125 as that resolution to beam width search resolves the issue that motivated this PR. Closing.
